### PR TITLE
BBS Animation Support & PMO Tools

### PR DESCRIPTION
**Pull request by Truthkey**
This PR includes several things:

Engine:

* Support for the PAM format (Birth by Sleep animations)
* Very superficial support for bones on Birth by Sleep Models
* Fixes to Vertex Color in PMP & PMO (Maps and Models)

Tools:

* Fully working FBX to PMO converter. Currently only supports static meshes.
* Prototype PAM to FBX converter (Not working at the moment)

### CODE REVIEW

File: OpenKh.Engine/Parsers/PmoParser.cs [hunk 1]
```diff
@@ -62,16 +50,63 @@ public PmoParser(Pmo pmo, float Scale)
...
+
+                    mtx_nd.M11 *= Scale;
+                    mtx_nd.M21 *= Scale;
+                    mtx_nd.M31 *= Scale;
+
+                    Vector3 loc;
+                    Quaternion quat;
+                    Vector3 scl;
+
+                    Matrix4x4.Decompose(mtx_nd, out scl, out quat, out loc);
```
Comment (Xeeynamo) [inline OpenKh.Engine/Parsers/PmoParser.cs:72]: I can see you're decomposing the matrix but you are not using `scl`, `quat` and `loc`. Is this line of code needed?

File: OpenKh.sln [hunk 2]
```diff
@@ -167,6 +167,7 @@ EndProject
 Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "OpenKh.Tools.ItbEditor", "OpenKh.Tools.ItbEditor\OpenKh.Tools.ItbEditor.csproj", "{0AC54BBF-A08D-44CA-9140-EB5C47573BAE}"
 EndProject
 Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "OpenKh.Tools.IteViewer", "IteEditor\OpenKh.Tools.IteViewer.csproj", "{7BA60C7F-4DA9-4A62-A1DC-0F8FF87E4430}"
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "OpenKh.Command.PAMtoFBXConverter", "OpenKh.Command.PAMtoFBXConverter\OpenKh.Command.PAMtoFBXConverter.csproj", "{A9695331-C335-4C62-B5B3-7FB3ED512EA3}"
```
Comment (Xeeynamo) [inline OpenKh.sln:170]: There is a missing `EndProject` between the two lines. This will break the solution file.

**Review by Xeeynamo — CHANGES_REQUESTED** [review]
There is a tiny request change I suggested before merging. Apart from that everything looks good 👍 . I can see that there is no `Pam.Write`, but I imagine it can easily come later. Very excited for all your amazing contribution Key 🥇

File: OpenKh.Engine/Parsers/PmoParser.cs (hunk 1 above)
Comment (Truthkey) [reply OpenKh.Engine/Parsers/PmoParser.cs:72]: Just removed it!

**Comment by Xeeynamo** [comment]
Maps are displayed flawlessly 🥺
image

File: OpenKh.sln (hunk 2 above)
Comment (Truthkey) [reply OpenKh.sln:171]: Thanks for the catch. I fixed it.

**Comment by Xeeynamo** [comment]
One test is failing:
```
OpenKh.Tests.Bbs.PamTests.ReadCorrectData
Assert.Equal() Failure
          ↓ (pos 0)
Expected: p01ex00001a
Actual:   OpenKHAnim
          ↑ (pos 0)
```

**Comment by Truthkey** [comment]
> One test is failing:
> 
> ```
> OpenKh.Tests.Bbs.PamTests.ReadCorrectData
> Assert.Equal() Failure
>           ↓ (pos 0)
> Expected: p01ex00001a
> Actual:   OpenKHAnim
>           ↑ (pos 0)
> ```

I fixed it.

**Review by Xeeynamo — APPROVED** [review]

**Pull request merged**